### PR TITLE
Devops: Improve testing

### PR DESCRIPTION
**Pull request by Lulalaby**
Follow-up of #98, by request of @mxschmitt

### CODE REVIEW

**Comment by Lulalaby** [comment]
Needs the discord token as secret, as send in dm

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,9 +1,10 @@
-name: CI
+name: "Test CI"
```
Comment (mxschmitt) [inline .github/workflows/ci.yml]:
```suggestion
name: Test
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
+
 on:
   push:
-    branches: [ master ]
```
Comment (mxschmitt) [inline .github/workflows/ci.yml:4]: let's bring that back?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -27,3 +28,5 @@ jobs:
         sudo ./startup
     - name: Run Tests
       run: go test -v startup_test.go
+      env:
+        DISCORD_KEY: ${{ secrets.SINUSBOT_DISCORD_CI }}
```
Comment (mxschmitt) [inline .github/workflows/ci.yml]:
```suggestion
        DISCORD_API_KEY: ${{ secrets.SINUSBOT_DISCORD_API_KEY }}
```

that makes it more clear

File: startup_test.go [hunk 4]
```diff
@@ -64,14 +94,36 @@ func TestIsBotOnTeamspeak(t *testing.T) {
...
+	}
+	token, err := login("admin", string(pw), *botId)
+	if err != nil {
+		t.Fatalf("could not get token: %v", err)
+	}
+	bots, err := getInstances(*token)
+	if err != nil {
+		t.Fatalf("could not get instances: %v", err)
+	}
+	if err := despawnInstance(bots[0].UUID, *token); err != nil {
```
Comment (mxschmitt) [inline startup_test.go]:
```suggestion
	if err := killInstance(bots[0].UUID, *token); err != nil {
```

File: startup_test.go [hunk 5]
```diff
@@ -64,14 +94,36 @@ func TestIsBotOnTeamspeak(t *testing.T) {
...
+		t.Fatalf("could not change instance settings: %v", err)
```
Comment (mxschmitt) [inline startup_test.go]:
```suggestion
		t.Fatalf("could not kill instance: %v", err)
```

File: startup_test.go [hunk 6]
```diff
@@ -64,14 +94,36 @@ func TestIsBotOnTeamspeak(t *testing.T) {
...
+	}
+	if err := despawnInstance(bots[1].UUID, *token); err != nil {
```
Comment (mxschmitt) [inline startup_test.go]: ditto

File: startup_test.go [hunk 7]
```diff
@@ -25,6 +27,34 @@ func TestIsBotRunning(t *testing.T) {
...
+	}
+	instance, err := createDiscordInstance(string(apitoken), *token)
+	if err != nil {
+		t.Fatalf("could not create instance: %v", err)
+	}
+	fmt.Println("Created instance ", instance.UUID)
+
+	if err := spawnInstance(instance.UUID, *token); err != nil {
+		t.Fatalf("could not spawn discord instance: %v", err)
+	}
```
Comment (mxschmitt) [inline startup_test.go:97]: we should assert here that the bot actually joined discord. So an API call to discord would make sense.

File: startup_test.go [hunk 8]
```diff
@@ -125,6 +214,38 @@ func changeSettings(uuid, token string) error {
 	return nil
 }
 
+func despawnInstance(uuid, token string) error {
+	req, err := http.NewRequest("POST", "http://127.0.0.1:8087/api/v1/bot/i/"+uuid+"/kill", nil)
```
Comment (mxschmitt) [inline startup_test.go]: a lot of similar stuff which happens in despawnInstance and spawnInstance etc. We should probably create an ApiClient wrapper.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (Lulalaby) [reply .github/workflows/ci.yml:4]: Contributors can't use the action on other branches when forked with it.
I don't exactly know why, but it shouldn't harm to leave that out?

File: startup_test.go (hunk 7 above)
Comment (Lulalaby) [reply startup_test.go:97]: Will do

**Comment by Lulalaby** [comment]
Due to the sinusbot bugs, we can't split this really. Both changes needs to be here

**Pull request closed without merging**